### DISCUSSION TITLE
Apache Reverse Proxy Doesn't work after update

### DISCUSSION THREAD

**Discussion by mgkeirug** [dani-garcia/vaultwarden#4377] [Q&A] [answered]
I updated my docker image after a year this morning and now I'm stuck unable to access Vaultwarden.  I have it running behind a reverse proxy on Apache and everything was working fine before today.  When I try to start the container:

`sudo docker run -d --name vaultwarden -v /srv/vw-data:/data -e WEBSOCKET_ENABLED=true -p 127.0.0.1:443:443 -p 127.0.0.1:5001:5001 --restart unless-stopped vaultwarden/server:latest`

I get the following error:

`docker: Error response from daemon: driver failed programming external connectivity on endpoint vaultwarden (f1235f2e28d80a4433efa07d84c8dc0f5e374968add942cb3cf22d54aae866e2): Error starting userland proxy: listen tcp4 127.0.0.1:443: bind: address already in use.`

I have my vhost setup as so:

        <VirtualHost *:443>

        ServerName vault.xxxxxx.net
                ProxyPreserveHost On
                ProxyRequests Off
                RewriteEngine On
                RewriteCond %{HTTP:Connection} Upgrade [NC]
                RewriteCond %{HTTP:Upgrade} websocket [NC]
                RewriteRule /notifications/hub(.*) ws://localhost:3012/$1 [P,L]
                ProxyPass / http://127.0.0.1:5001/
                ProxyPassReverse / http://127.0.0.1:5001/

        <Location />
                Require ip xxx.xxx
                Require ip xxx.xxx.xxx.xxx
        </Location>

        ErrorLog ${APACHE_LOG_DIR}/vault_error.log
        CustomLog ${APACHE_LOG_DIR}/vault_access.log combined

        Include /etc/letsencrypt/options-ssl-apache.conf

        SSLCertificateFile /etc/letsencrypt/live/xxxxxx/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/xxxxxx/privkey.pem

        </VirtualHost>

After some research, I just removed the 443 port and websocket when starting the container as so:

`sudo docker run -d --name vaultwarden -v /srv/vw-data:/data -p 127.0.0.1:5001:5001 --restart unless-stopped vaultwarden/server:latest`

and altered the vhost to include upgrade=websocket:

        <VirtualHost *:443>
        ServerName vault.xxxxxx.net
                ProxyPreserveHost On
                ProxyRequests Off
                RewriteEngine On
                RewriteCond %{HTTP:Connection} Upgrade [NC]
                RewriteCond %{HTTP:Upgrade} websocket [NC]
                #RewriteRule /notifications/hub(.*) ws://localhost:3012/$1 [P,L]
                ProxyPass / http://127.0.0.1:5001/ upgrade=websocket
                ProxyPassReverse / http://127.0.0.1:5001/

        <Location />
                Require ip xxx.xxx
                Require ip xxx.xxx.xxx.xxx
        </Location>

        ErrorLog ${APACHE_LOG_DIR}/vault_error.log
        CustomLog ${APACHE_LOG_DIR}/vault_access.log combined

        Include /etc/letsencrypt/options-ssl-apache.conf

        SSLCertificateFile /etc/letsencrypt/live/xxxxxx/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/xxxxxx/privkey.pem
        </VirtualHost>

Now it will start properly but I still get proxy errors in the apache error log:

> [Thu Feb 22 17:06:23.281119 2024] [proxy_http:error] [pid 2603609] (104)Connection reset by peer: [client xxx.xxx.xxx.xxx:58610] AH01102: error reading status line from remote server 127.0.0.1:5001
> [Thu Feb 22 17:06:23.283996 2024] [proxy:error] [pid 2603609] [client xxx.xxx.xxx.xxx:58610] AH00898: Error reading from remote server returned by /
> [Thu Feb 22 17:06:23.340626 2024] [proxy_http:error] [pid 2603609] (104)Connection reset by peer: [client xxx.xxx.xxx.xxx:58610] AH01102: error reading status line from remote server 127.0.0.1:5001, referer: https://vault.xxxxxx.net/
> [Thu Feb 22 17:06:23.430745 2024] [proxy_http:error] [pid 2603608] (20014)Internal error (specific information not available): [client xxx.xxx.xxx.xxx:58612] AH01102: error reading status line from remote server 127.0.0.1:5001, referer: https://vault.xxxxxx.net/
> [Thu Feb 22 17:06:23.430826 2024] [proxy:error] [pid 2603608] [client xxx.xxx.xxx.xxx:58612] AH00898: Error reading from remote server returned by /favicon.ico, referer: https://vault.xxxxxx.net/

Other note when I run:
`sudo lsof -i -P -n |grep LISTEN`

It does list:
`docker-pr 2602983            root    4u  IPv4 13584565      0t0  TCP 127.0.0.1:5001 (LISTEN)`

And when I run:
`docker ps`

It lists:
```
CONTAINER ID   IMAGE                                 COMMAND                  CREATED          STATUS                    PORTS                                        NAMES
fb59b0081574   vaultwarden/server:latest             "/start.sh"              16 minutes ago   Up 16 minutes (healthy)   80/tcp, 3012/tcp, 127.0.0.1:5001->5001/tcp   vaultwarden
```

I haven't messed around with this for a year and I don't know if something changed that would cause this if its just me.

**Comment by stefan0xC** [comment, marked as answer]
> `sudo docker run -d --name vaultwarden -v /srv/vw-data:/data -p 127.0.0.1:5001:5001 --restart unless-stopped vaultwarden/server:latest`

The container image is setup to start Vaultwarden on port `80` (so you either want to expose the port like this `-p 127.0.0.1:5001:80` or alternatively, you could change the port by setting `ROCKET_PORT=5001`).

**Reply by mgkeirug** [reply]
That did it!  Thank you so much!
